src/tools/time_governor.cc

```diff
@@ -38,37 +43,101 @@ const double TimeGovernor::time_step_precision = 16*numeric_limits<double>::epsi
 using namespace Input::Type;
 
 
+const Tuple & TimeGovernor::get_input_time_type(double lower_bound, double upper_bound)
+{
+    return Tuple("TimeValue", "Value of Field for independent variable.")
+        .declare_key("time", Double(lower_bound, upper_bound), Default::obligatory(),
+                                    "Numeric value of time." )
+		.declare_key("unit", String(), Default::optional(),
+									"Specify unit of an input time value. This value overrides default unit "
+									"of equation specified by 'common_time_unit' key of Time Governor.")
+		.close();
+}
+
+
 const Record & TimeGovernor::get_input_type() {
 	return Record("TimeGovernor",
             "Setting of the simulation time. (can be specific to one equation)")
 		.allow_auto_conversion("max_dt")
-		.declare_key("start_time", Double(), Default("0.0"),
+		.declare_key("start_time", TimeGovernor::get_input_time_type(), Default("0.0"),
 					"Start time of the simulation.")
-		.declare_key("end_time", Double(), Default(MAX_END_TIME_STR),
+		.declare_key("end_time", TimeGovernor::get_input_time_type(), Default(MAX_END_TIME_STR),
 					"End time of the simulation. Default value is more then age of universe in seconds.")
-		.declare_key("init_dt", Double(0.0), Default("0.0"),
+		.declare_key("init_dt", TimeGovernor::get_input_time_type(0.0), Default("0.0"),
 				"Initial guess for the time step.\n"
 				"Only useful for equations that use adaptive time stepping."
 				"If set to 0.0, the time step is determined in fully autonomous"
 				" way if the equation supports it.")
-		.declare_key("min_dt", Double(0.0),
+		.declare_key("min_dt", TimeGovernor::get_input_time_type(0.0),
 				Default::read_time("Machine precision."),
 				"Soft lower limit for the time step. Equation using adaptive time stepping can not"
 				"suggest smaller time step, but actual time step could be smaller in order to match "
 				"prescribed input or output times.")
-		.declare_key("max_dt", Double(0.0),
+		.declare_key("max_dt", TimeGovernor::get_input_time_type(0.0),
 				Default::read_time("Whole time of the simulation if specified, infinity else."),
 				"Hard upper limit for the time step. Actual length of the time step is also limited"
 				"by input and output times.")
+		.declare_key("common_time_unit", String(), Default("\"s\""),
+				"Common time unit of equation. This unit will be used for all time inputs and outputs "
```

Comment:
Add explicit list of time influenced time units:
In particular:
1. time units of TimeValue keys in: TimeGovernor, EquationTimeSet, ... 
2. time units in input fields: FieldElementwise, FieldTimeFunction, ...
3. time units in output files: Observe times, balance times, frame times of VTK and GMSH field output